src/cdk/v2/destinations/rakuten/utils.js

```diff
@@ -17,6 +17,18 @@ const constructProperties = (message) => {
   return payload;
 };
 
+/**
+ * Calculates the amount for a single product
+ * @param {Object} product
+ * @returns {number}
+ */
+const calculateProductAmount = (product) => {
+  if (!product?.amount && !product?.price) {
+    throw new InstrumentationError('Either amount or price is required for every product');
+  }
+  return Math.round(product.amount) * 100 || (product.quantity || 1) * 100 * product.price;
```

Comment:
```suggestion
  return Math.round(product.amount) * 100 || (product.quantity || 1) * Math.round(100 * product.price);
```
Should we also do ?

Reply:
return Math.round(product.amount * 100 || (product.quantity || 1) * 100 * product.price);

Reply:
we should do Math.round after multiplying with 100 right?

Reply:
yes

Reply:
yes yes, changed